agent/command.go

```diff
@@ -66,14 +64,15 @@ func (a *Agent) runCommands(ctx context.Context, tc *taskContext, commands []mod
 
 			if isTaskCommands {
 				tc.setCurrentCommand(cmd)
-				a.updateIdleTimeout(ctx, tc, timeoutPeriod, idleTimeout)
+				tc.setCurrentTimeout(a.getTimeout(commandInfo))
+				a.comm.UpdateLastMessageTime()
 			}
 
 			start := time.Now()
 			err = cmd.Execute(ctx, a.comm, tc.logger, tc.taskConfig)
+			tc.setCurrentTimeout(0)
```

Comment:
This resets what was set on line 67.

Reply:
I think that's desireable? once the command executes, then the next command should set the timeout (we could also set it to default at this point.